platform/macos/display_server_embedded.mm

```diff
@@ -0,0 +1,744 @@
+	window_event_callbacks[p_window] = p_callable;
+}
+void DisplayServerEmbedded::window_set_input_event_callback(const Callable &p_callable, WindowID p_window) {
+	input_event_callbacks[p_window] = p_callable;
+}
+
+void DisplayServerEmbedded::window_set_input_text_callback(const Callable &p_callable, WindowID p_window) {
+	input_text_callbacks[p_window] = p_callable;
+}
+
+void DisplayServerEmbedded::window_set_drop_files_callback(const Callable &p_callable, WindowID p_window) {
+	// Not supported
+}
+
+void DisplayServerEmbedded::joy_add(int p_idx, const String &p_name) {
+	Joy *joy = joysticks.getptr(p_idx);
+	if (joy == nullptr) {
+		joysticks[p_idx] = Joy(p_name);
+		Input::get_singleton()->joy_connection_changed(p_idx, true, p_name);
+	}
+}
+
+void DisplayServerEmbedded::joy_del(int p_idx) {
+	if (joysticks.erase(p_idx)) {
+		Input::get_singleton()->joy_connection_changed(p_idx, false, String());
+	}
+}
+
+void DisplayServerEmbedded::process_events() {
+	Input *input = Input::get_singleton();
+	for (KeyValue<int, Joy> &kv : joysticks) {
+		uint64_t ts = input->get_joy_vibration_timestamp(kv.key);
+		if (ts > kv.value.timestamp) {
+			kv.value.timestamp = ts;
+			Vector2 strength = input->get_joy_vibration_strength(kv.key);
+			if (strength == Vector2()) {
+				EngineDebugger::get_singleton()->send_message("game_view:joy_stop", { kv.key });
+			} else {
+				float duration = input->get_joy_vibration_duration(kv.key);
+				EngineDebugger::get_singleton()->send_message("game_view:joy_start", { kv.key, duration, strength });
+			}
+		}
+	}
+
+	input->flush_buffered_events();
+}
+
+void DisplayServerEmbedded::_dispatch_input_events(const Ref<InputEvent> &p_event) {
+	Ref<InputEventFromWindow> event_from_window = p_event;
+	WindowID window_id = INVALID_WINDOW_ID;
+	if (event_from_window.is_valid()) {
+		window_id = event_from_window->get_window_id();
+	}
+	DisplayServerEmbedded *ds = (DisplayServerEmbedded *)DisplayServer::get_singleton();
+	ds->send_input_event(p_event, window_id);
+}
+
+void DisplayServerEmbedded::send_input_event(const Ref<InputEvent> &p_event, WindowID p_id) const {
+	if (p_id != INVALID_WINDOW_ID) {
+		_window_callback(input_event_callbacks[p_id], p_event);
+	} else {
+		for (const KeyValue<WindowID, Callable> &E : input_event_callbacks) {
+			_window_callback(E.value, p_event);
+		}
+	}
+}
+
+void DisplayServerEmbedded::send_input_text(const String &p_text, WindowID p_id) const {
+	const Callable *cb = input_text_callbacks.getptr(p_id);
+	if (cb) {
+		_window_callback(*cb, p_text);
+	}
+}
+
+void DisplayServerEmbedded::send_window_event(DisplayServer::WindowEvent p_event, WindowID p_id) const {
+	const Callable *cb = window_event_callbacks.getptr(p_id);
+	if (cb) {
+		_window_callback(*cb, int(p_event));
+	}
+}
+
+void DisplayServerEmbedded::_window_callback(const Callable &p_callable, const Variant &p_arg) const {
+	if (!p_callable.is_null()) {
```

Comment:
```suggestion
	if (p_callable.is_valid()) {
```
Please use `is_valid` for checking if a `Callable` can be called, `is_null` is not equivalent